NMS-14992: Update Inventory and Performance Data Collection QS chapters (#5518)
OpenNMS/opennms: docs/modules/operation/pages/quick-start/inventory.adoc

= Step 3: Provision Inventory

{page-component-title} installs with a base configuration that automatically begins service-level monitoring and reporting as soon as you identify and catalog (provision) the managed entities (nodes, IP interfaces, SNMP interfaces, and services) that you want to monitor.
The default configuration also enables self-monitoring of the OpenNMS service, and discovers a single node with the interface `127.0.0.0` (see xref:deployment:core/getting-started.adoc#first-monitored-node[First monitored node]).

{page-component-title} has three methods of provisioning your network's inventory: xref:deep-dive/provisioning/auto-discovery.adoc[automatic discovery], xref:deep-dive/provisioning/directed-discovery.adoc[directed discovery], and xref:development:rest/rest-api.adoc[through the REST API].

Regardless of the method, provisioning is an iterative process.
You will need to fine-tune your results to exclude or add to what you monitor.

This Quick Start guide presents a very basic example using automatic discovery to provision nodes through SNMP.

== SNMP in {page-component-title}

> SNMP is used in {page-component-title} to understand network and node topology, collect performance data, and enable SNMP service monitoring of nodes.

Reviewer: ```suggestion
{page-component-title} uses SNMP to understand network and node topology, collect performance data, and enable SNMP service monitoring of nodes.
```

Author: I'm not sure... @mhuot, would you be able to clarify?

SNMP service monitoring in particular ensures that SNMP is available on a given node; it checks a configurable OID value to understand the status of a particular part, function, or other aspect of the monitored device.
Tracking the state of this value can be used to trigger a notification if it represents an error or another condition that may need attention.

> The service also gathers basic information about a monitored device, including the device type, its name, its location, and so on.

Author: Because we use "and so on" in other lists like this, I think it's okay to be kept as is.

[[provision-snmp-configuration]]
== Configure SNMP for provisioning

To collect data through SNMP or monitor the availability of SNMP services on a node, you need to configure SNMP for provisioning.
This ensures that {page-component-title} can immediately scan newly discovered devices for entities.
{page-component-title} updates the network topology--the visual representation of the monitored network and its components--as it provisions nodes.

This procedure illustrates a very basic SNMP configuration using the default parameters.
You can customize these values for your own network needs, although the default settings are sufficient for a basic {page-component-title} setup:

. Click the *gear* symbol in the top-right of the screen.
. Under Provisioning, click *Configure SNMP Community Names by IP Address*.
. Configure the community names as appropriate for your environment:
** Enter a valid IP address in the *IP Address* box, and click *Look Up*.
** Choose an SNMP version (`v2c` or `v1`) from the *Version* list.
** (Optional) Define a range of IP addresses by typing the first and last addresses of the range into the *First IP Address* and *Last IP Address* boxes, respectively.

> ** In the v1/v2c Specific Parameters section, enter the new community string in the *Read Community String* box.

Reviewer: Is there anything they need to set for Write Community string? Including the Read and Write was something Mike had requested.

Reviewer 2: I think this would also be good to clarify.  I don't believe we write anything by default.  Additionally, this section mentions that SNMPv3 is beyond the scope and I'm not sure it should be.  I think it would be good to cover it here as it's not an uncommon thing someone would want help with getting started on.

Author: Thanks! We have a task to update the SNMP documentation as it is, so this will be covered in that ticket (NMS-12873).

[[provision-snmp-configuration]]
== Configure SNMP for provisioning

To collect data through SNMP or monitor the availability of SNMP services on a node, you need to configure SNMP for provisioning.
This ensures that {page-component-title} can immediately scan newly discovered devices for entities.
{page-component-title} updates the network topology--the visual representation of the monitored network and its components--as it provisions nodes.

This procedure illustrates a very basic SNMP configuration using the default parameters.
You can customize these values for your own network needs, although the default settings are sufficient for a basic {page-component-title} setup:

. Click the *gear* symbol in the top-right of the screen.
. Under Provisioning, click *Configure SNMP Community Names by IP Address*.
. Configure the community names as appropriate for your environment:
** Enter a valid IP address in the *IP Address* box, and click *Look Up*.
** Choose an SNMP version (`v2c` or `v1`) from the *Version* list.
+
NOTE: If you are using SNMPv3, you need to configure details for a User-Based Security Model (USM) user instead of a community string.

> {page-component-title} supports SNMPv3 and USM, but these topics are outside of the scope of this guide.

Reviewer: ```suggestion
{page-component-title} supports SNMPv3 and USM, but these topics are outside of the scope of this Quick Start guide.
```
Just to make it clear we don't mean the core docs. We will document this soon!

Author: I'll note this for cross-reference once we document it. :)

> ** (Optional) Define a range of IP addresses by typing the first and last addresses of the range into the *First IP Address* and *Last IP Address* boxes, respectively.

Reviewer: ```suggestion
** (Optional) Define a range of IP addresses by typing the first and last address of the range into the *First IP Address* and *Last IP Address* boxes, respectively.
```
